enhancements/monitoring/alert-routing-for-user-workload-monitoring.md

```diff
@@ -0,0 +1,417 @@
+monitored by the user-workload monitoring without requesting intervention from
+cluster admins.
+* Cluster admins can grant permissions to users and groups to manage alert
+routing scoped to individual namespaces.
+* Namespace owners should be able to opt-out from Alertmanager
+configuration (similar to what exist for service/pod monitors and rules using the
+`"openshisft.io/user-monitoring: false"` label on the namespace).
+* Cluster admins should be able to opt-out from supporting `AlertmanagerConfig`
+resources from user namespaces.
+
+### Non-Goals
+
+* Additional support for silencing user alerts (it is already supported by UWM in the OCP console).
+* Specific integration in the OCP Console exposing the configuration of alert notifications.
+* Support the configuration of alert notifications for platform alerts (e.g.
+alerts originating from namespaces with the `openshift.io/cluster-monitoring: "true"`
+label).
+
+## Proposal
+
+We plan to leverage the `AlertmanagerConfig` custom resource definition already
+exposed by the Prometheus operator so that application owners can configure how
+and where their alert notifications should be routed.
+
+### User Stories
+
+Personas:
+* Application owners: manage a project with sufficient permissions to define monitoring resources.
+* UWM admins: manage the configuration of the UWM components (edit permissions on the `openshift-user-workload-monitoring/user-workload-monitoring-config` configmap).
+* Cluster admins: manage the configuration of the Platform monitoring components.
+
+#### Story 1
+
+As an application owner, I want to use AlertmanagerConfig custom resources so
+that Alertmanager can push alert notifications for my applications to the
+receiver of my choice.
+
+#### Story 2
+
+As an application owner, I want to use AlertmanagerConfig custom
+resources so that Alertmanager can inhibit alerts based on other alerts firing
+at the same time.
+
+#### Story 3
+
+As an application owner, I want to know if my AlertmanagerConfig custom
+resource is taken into account so that I am confident that I will receive alert
+notifications.
+
+#### Story 4
+
+As a OpenShift cluster admin, I want to allow some of my users to
+create/update/delete AlertmanagerConfig custom resources and leverage the
+platform Alertmanager cluster so that I don't have to configure alert routing
+on their behalf.
+
+#### Story 5
+
+As an OpenShift cluster admin, I don't want AlertmanagerConfig resources
+defined by application owners to interfere with the routing of platform alerts.
+
+#### Story 6
+
+As an OpenShift cluster admin, I want to exclude certain user namespaces from
+modifying the Plaform Alertmanager configuration so that I can recover in case
+of breakage or bad behavior.
+
+#### Story 7
+
+As an OpenShift cluster admin, I don't want to support AlertmanagerConfig
+resources for application owners so that the configuration of the Platform
+Alertmanager cluster is completely under my control.
+
```

Comment:
Does this equate to full opt-out of the feature at a cluster wide level?

Reply:
yes

Reply:
The CRD will always exist but at least the Platform Alertmanager won't gather any user `AlertmanagerConfig`s.